### ISSUE TITLE
Stream start - wrong channel data output

### ISSUE THREAD

**Issue by keynet** [carsten-gross/ts2shout#2]
Hello again. 
I've been using ts2shout daily for a couple of weeks now, it's working well, but I've observed an occasional problem with stream startups. Below is an annotated example from a stream start this morning.  Data is always correct (and doesn't go wrong) once the stream is fully established.  It appears data from another stream is being output during the stream startup process.

```
(Correct) 14/04/2020 08:55:00 album/stream=BBC Radio 4 (Freeview), artist/title=Today, track/desc=News and current affairs, including Sports Desk, Weather and Thought for the Day
(Partial, but correct) 14/04/2020 08:54:47 album/stream=BBC Radio 4 (Freeview), track/desc=BBC Radio 4
(Data from a different stream + garbage) 4/04/2020 08:54:47 album/stream=BBC Radio 4 (Freeview), artist/title=The Repair Shop, track/desc=3/14. Jay Blades and the team tackle a torn leather armchair, a broken music box, a faded portrait and a shattered stained-glass door panel. Also in HD. [S,AD,SL | #:D@+|||||P|||||undP|||||engP|||G|engP|||||engP|||0|engT|
(start) 4/04/2020 08:54:46 album/stream=BBC Radio 4 (Freeview), track/desc=BBC Radio 4 (Freeview)

Here is the Apache2 log

```
[Tue Apr 14 08:54:45.852877 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:45.852675 2020] [ts2shout:info] [pid 22927] init_structures(): Subscribing to MPEG-TS PID 0, 17, 18 (CHANNEL_TYPE_PAT, CHANNEL_TYPE_SDT, CHANNEL_TYPE_EIT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:45.852983 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:45.852718 2020] [ts2shout:info] [pid 22927] ts2shout version v0.88-3-ga66bfb1 compiled 04.04.2020 13:15 started: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:45.853064 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:45.852725 2020] [ts2shout:info] [pid 22927] MPEG streaming with shoutcast StreamTitles in CGI mode with RDS support.: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:45.853145 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:45.852842 2020] [ts2shout:info] [pid 22927] fetch_cached_parameters(): found parameters for programme T-9018-4164-6912: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.058239 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.058080 2020] [ts2shout:info] [pid 22927] extract_pat_payload(): programme has transport_stream_id: 4164: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.058368 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.058104 2020] [ts2shout:info] [pid 22927] add_channel(): Subscribing to MPEG-TS PID 1300 (Type CHANNEL_TYPE_PMT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.066929 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.066728 2020] [ts2shout:info] [pid 22932] init_structures(): Subscribing to MPEG-TS PID 0, 17, 18 (CHANNEL_TYPE_PAT, CHANNEL_TYPE_SDT, CHANNEL_TYPE_EIT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.067038 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.066776 2020] [ts2shout:info] [pid 22932] ts2shout version v0.88-3-ga66bfb1 compiled 04.04.2020 13:15 started: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.067119 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.066784 2020] [ts2shout:info] [pid 22932] MPEG streaming with shoutcast StreamTitles in CGI mode with RDS support.: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.067200 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.066892 2020] [ts2shout:info] [pid 22932] fetch_cached_parameters(): found parameters for programme T-9018-4164-6912: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.158428 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.158245 2020] [ts2shout:info] [pid 22927] EIT: Current transmission `Breakfast on BBC Essex': /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.158545 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.158288 2020] [ts2shout:info] [pid 22927] add_payload_from_pmt(): Found mp1/mp2 audio stream in PID 1302 (service_id 6912): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.158636 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.158296 2020] [ts2shout:info] [pid 22927] add_channel(): Subscribing to MPEG-TS PID 1302 (Type CHANNEL_TYPE_PAYLOAD): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.171821 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.171686 2020] [ts2shout:info] [pid 22932] extract_pat_payload(): programme has transport_stream_id: 4164: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.171922 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.171709 2020] [ts2shout:info] [pid 22932] add_channel(): Subscribing to MPEG-TS PID 1300 (Type CHANNEL_TYPE_PMT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.258501 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53100] AH01215: [Tue Apr 14 08:54:46.258322 2020] [ts2shout:info] [pid 22927] Synced to MPEG-1 layer 2, 192 kbps, 48000 Hz, Joint Stereo: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.271904 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.271772 2020] [ts2shout:info] [pid 22932] EIT: Current transmission `The Repair Shop': /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.272010 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.271813 2020] [ts2shout:info] [pid 22932] add_payload_from_pmt(): Found mp1/mp2 audio stream in PID 1302 (service_id 6912): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.272094 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.271820 2020] [ts2shout:info] [pid 22932] add_channel(): Subscribing to MPEG-TS PID 1302 (Type CHANNEL_TYPE_PAYLOAD): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:46.372116 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:46.371976 2020] [ts2shout:info] [pid 22932] Synced to MPEG-1 layer 2, 192 kbps, 48000 Hz, Joint Stereo: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.278222 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.278005 2020] [ts2shout:info] [pid 22938] init_structures(): Subscribing to MPEG-TS PID 0, 17, 18 (CHANNEL_TYPE_PAT, CHANNEL_TYPE_SDT, CHANNEL_TYPE_EIT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.278350 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.278262 2020] [ts2shout:info] [pid 22938] ts2shout version v0.88-3-ga66bfb1 compiled 04.04.2020 13:15 started: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.278454 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.278367 2020] [ts2shout:info] [pid 22938] MPEG streaming with shoutcast StreamTitles in CGI mode with RDS support.: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.278657 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.278563 2020] [ts2shout:info] [pid 22938] fetch_cached_parameters(): found parameters for programme T-9018-4164-6912: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.383669 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.383552 2020] [ts2shout:info] [pid 22938] extract_pat_payload(): programme has transport_stream_id: 4164: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.383797 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.383701 2020] [ts2shout:info] [pid 22938] add_channel(): Subscribing to MPEG-TS PID 1300 (Type CHANNEL_TYPE_PMT): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.474067 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:47.473858 2020] [ts2shout:info] [pid 22932] SDT: Stream is station BBC Radio 4 from network .: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.483704 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.483580 2020] [ts2shout:info] [pid 22938] add_payload_from_pmt(): Found mp1/mp2 audio stream in PID 1302 (service_id 6912): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.483824 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.483729 2020] [ts2shout:info] [pid 22938] add_channel(): Subscribing to MPEG-TS PID 1302 (Type CHANNEL_TYPE_PAYLOAD): /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:47.583814 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:47.583697 2020] [ts2shout:info] [pid 22938] Synced to MPEG-1 layer 2, 192 kbps, 48000 Hz, Joint Stereo: /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:48.975949 2020] [cgi:error] [pid 5967] [client 10.0.0.100:53104] AH01215: [Tue Apr 14 08:54:48.975839 2020] [ts2shout:info] [pid 22932] EIT: Current transmission `Today': /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:48.985767 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:48.985681 2020] [ts2shout:info] [pid 22938] EIT: Current transmission `Today': /srv/www/cgi-bin/ts2shout
[Tue Apr 14 08:54:49.487010 2020] [cgi:error] [pid 14721] [client 10.0.0.100:53108] AH01215: [Tue Apr 14 08:54:49.486872 2020] [ts2shout:info] [pid 22938] SDT: Stream is station BBC Radio 4 from network .: /srv/www/cgi-bin/ts2shout
```



```

**Comment by carsten-gross** [comment]
Hello keynet,

Am Tue, 14 Apr 2020, schrieb keynet:
> I've been using ts2shout daily for a couple of weeks now, it's working well, but I've observed an occasional problem with stream startups. Below is an annotated example from a stream start this morning.  Data is always correct (and doesn't go wrong) once the stream is fully established.  It appears data from another stream is being output during the stream startup process.

Could you please send me the output of ts2shout (written into the
apache error-log). Data from another stream (inside the multiplex) could be
possible and would be a bug. Old data (from a run of the software earlier) is very
unlikely (because ts2shout just reads from stdin or makes a "fresh"
fetch with libcurl). 

> (Correct) 14/04/2020 08:55:00 album/stream=BBC Radio 4 (Freeview), artist/title=Today, track/desc=News and current affairs, including Sports Desk, Weather and Thought for the Day
> (Partial, but correct) 14/04/2020 08:54:47 album/stream=BBC Radio 4 (Freeview), track/desc=BBC Radio 4
> (Data from a different stream + garbage) 4/04/2020 08:54:47 album/stream=BBC Radio 4 (Freeview), artist/title=The Repair Shop, track/desc=3/14. Jay Blades and the team tackle a torn leather armchair, a broken music box, a faded portrait and a shattered stained-glass door panel. Also in HD. [S,AD,SL | #:D@+|||||P|||||undP|||||engP|||G|engP|||||engP|||0|engT|
> (start) 4/04/2020 08:54:46 album/stream=BBC Radio 4 (Freeview), track/desc=BBC Radio 4 (Freeview)

Viele Grüße,

   Carsten Groß

-- 
Carsten Gross            | http://www.siski.de/~carsten/

**Comment by carsten-gross** [comment]
Pre-stream-start EIT processing fixed in commit 5e349d9
